### PR TITLE
Fix issue where polymer-cli throws when help flag is provided alone

### LINKED ISSUE #189: Unknown option throws error
Invoking `polymer --help` threw the following error:

```
error:   Uncaught exception: UNKNOWN_OPTION: Unknown option: --help
error:   UNKNOWN_OPTION: Unknown option: --help
    at halt (/usr/local/lib/node_modules/polymer-cli/node_modules/command-line-args/lib/argv.js:81:15)
    at Argv.validate (/usr/local/lib/node_modules/polymer-cli/node_modules/command-line-args/lib/argv.js:72:7)
    at CommandLineArgs.parse (/usr/local/lib/node_modules/polymer-cli/node_modules/command-line-args/lib/command-line-args.js:57:10)
    at CommandLineCommands.parse (/usr/local/lib/node_modules/polymer-cli/node_modules/command-line-commands/lib/command-line-commands.js:78:28)
    at PolymerCli.run (/usr/local/lib/node_modules/polymer-cli/lib/polymer-cli.js:122:35)
    at /usr/local/lib/node_modules/polymer-cli/bin/polymer.js:22:7
    at /usr/local/lib/node_modules/polymer-cli/node_modules/resolve/lib/async.js:46:14
    at process (/usr/local/lib/node_modules/polymer-cli/node_modules/resolve/lib/async.js:173:43)
    at ondir (/usr/local/lib/node_modules/polymer-cli/node_modules/resolve/lib/async.js:188:17)
    at load (/usr/local/lib/node_modules/polymer-cli/node_modules/resolve/lib/async.js:69:43)
```

I expect to either not throw an error and gracefully stop or showing the help as fallback.

### PR DESCRIPTION
**Pull request by FredKSchott**
Fixes #189. Also gets rid of the last, now-unnecessary manual args parsing and replaces it with our more robust global-argument handling.

/cc @justinfagnani

### CODE REVIEW

File: src/args.ts [hunk 1]
```diff
@@ -55,6 +55,13 @@ export const globalArguments: ArgDescriptor[] = [
     group: 'global',
   },
   {
+    name: 'help',
+    description: 'silence output',
```
Comment (justinfagnani) [inline src/args.ts]: bad description :)

File: src/polymer-cli.ts [hunk 2]
```diff
@@ -132,26 +132,30 @@ export class PolymerCli {
...
+      && cliCommand.options['_all']);
+    logger.debug('command parsed', { name: commandName, command: cliCommand });
+    logger.debug('command options found', { options: commandOptions });
+
+    // When neccessary user input (incl. a sub-command name) is missing,
+    // make sure the help command runs as accurately as possible.
+    if (!commandName) {
+      commandOptions = {};
+      commandName = 'help';
+    } else if (!commandOptions || commandOptions['help']) {
```
Comment (justinfagnani) [inline src/polymer-cli.ts]: In what cases do we just not have a commandOptions object? What about commands that don't need options and maybe don't have defaults, do we have any of those?

File: test/commands/cli_test.js [hunk 3]
```diff
@@ -27,7 +27,16 @@ suite('The general CLI', () => {
     let helpCommandSpy = sinon.spy(helpCommand, 'run');
     cli.run();
     assert.isOk(helpCommandSpy.calledOnce);
-    assert.deepEqual(helpCommandSpy.firstCall.args, [undefined, defaultConfig]);
+    assert.deepEqual(helpCommandSpy.firstCall.args, [{}, defaultConfig]);
+  });
+
+  test('displays general help when no command is called with the --help flag', () => {
+    let cli = new PolymerCli(['--help']);
+    let helpCommand = cli.commands.get('help');
```
Comment (justinfagnani) [inline test/commands/cli_test.js]: ugh... we should refactor PolymerCli to be testable w/o monkey patching

**Comment by justinfagnani** [comment]
Depending on the answer to my question about when we don't get a commandOptions object, LGTM

**Comment by justinfagnani** [comment]
Oh, but rebase on master and let the bots run

File: src/args.ts (hunk 1 above)
Comment (FredKSchott) [inline src/args.ts]: whoops! done

File: src/polymer-cli.ts (hunk 2 above)
Comment (FredKSchott) [inline src/polymer-cli.ts]: You're right, `commandOptions` is always an object. Here's an example when no options are passed:
```
$ polymer init
console.log(cliCommand.options); // { _all: {}, global: {} }
console.log(commandOptions); // {}
```

I was just including that to be safe before attempting to access the help command. I'd usually keep it to protect from throwing an ugly error if our dependencies change behavior on us, but if you have strong feelings I'm happy to remove.

Comment (FredKSchott) [inline src/polymer-cli.ts]: You know what, I'll do us one better. Changing to 
```
if (commandOptions && commandOptions['help'])`
```

File: test/commands/cli_test.js (hunk 3 above)
Comment (FredKSchott) [inline test/commands/cli_test.js]: Where do you see monkey patching here? Are you talking about sinon?

**Pull request merged**